### PR TITLE
Automatically pull test reports from CI

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118931
* __->__ #118882
* #118874

### CODE REVIEW

File: scripts/compile_tests/failures_histogram.py [hunk 1]
```diff
@@ -149,6 +146,5 @@ def as_issue(count, msg, repro, tests=None):
         action="store_true",
     )
     args = parser.parse_args()
-    failures_histogram(
-        args.eager_dir, args.dynamo_dir, args.verbose, args.format_issues
-    )
+    dynamo38, dynamo311, eager311 = download_reports(args.commit)
```
Comment (jamesjwu) [inline scripts/compile_tests/failures_histogram.py:153]: do we not need eager38? There are likely tests that are skipped on eager3.8 that are also skipped on dynamo3.8 (esp. numpy ones), but pass on eager3.11.

File: scripts/compile_tests/update_failures.py [hunk 2]
```diff
@@ -173,9 +168,7 @@ def read_test_results(directory):
     )
     # dynamo_test_failures path
     parser.add_argument("filename")
-    # linux-focal-py3.8-clang10 (dynamo) Test Reports (xml) directory
-    parser.add_argument("py38_test_reports_dir")
-    # linux-focal-py3.11-clang10 (dynamo) Test Reports (xml) directory
-    parser.add_argument("py311_test_reports_dir")
+    parser.add_argument("commit")
```
Comment (jamesjwu) [inline scripts/compile_tests/update_failures.py:174]: Add instructions 
```python
parser.add_argument("commit", help="The commit sha for the latest commit on a PR from which we will pull CI test results, e.g. <ex>")
```

File: scripts/compile_tests/failures_histogram.py [hunk 3]
```diff
@@ -136,10 +136,7 @@ def as_issue(count, msg, repro, tests=None):
         prog="failures_histogram",
         description="See statistics about skipped Dynamo tests",
     )
-    # linux-focal-py3.11-clang10 (default) Test Reports (xml) directory
-    parser.add_argument("eager_dir")
-    # linux-focal-py3.11-clang10 (dynamo) Test Reports (xml) directory
-    parser.add_argument("dynamo_dir")
+    parser.add_argument("commit")
```
Comment (jamesjwu) [inline scripts/compile_tests/failures_histogram.py:143]: Same here

File: scripts/compile_tests/download_reports.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        job_id = workflow_jobs[job_name]
+        for listing in listings:
+            name = listing['name']
+            if not name.startswith("test-reports-"):
+                continue
+            if name.endswith(f'_{job_id}.zip'):
+                url = listing['url']
+                subprocess.run(["wget", "-P", subdir, url], check=True)
+                path_to_zip = f"{subdir}/{name}"
+                subprocess.run(["unzip", path_to_zip, "-d", path_to_zip[:-4]], check=True)
```
Comment (jamesjwu) [inline scripts/compile_tests/download_reports.py]: nit: [:-4] is just removing `.zip` right? Maybe add
dir_name = path_to_zip[:-4] 
and use that variable here for clarity

**Review by jamesjwu — APPROVED** [review]
It'd be nice to be able to just say "main" and have it pull the commit from origin/main or origin/viable/strict or whatever has CI finished

File: scripts/compile_tests/failures_histogram.py (hunk 1 above)
Comment (zou3519) [reply scripts/compile_tests/failures_histogram.py:153]: I'll figure out how to do this in a follow-up. The failures_histogram.py script only worked on a single python version before, but now that we have a way to more easily download reports we should actually do something with the python 3.8 information.

**Review by yanboliang — APPROVED** [review]
Great productivity tool!

**Pull request closed without merging**